drivers/storage/partmgr/partition.c

```diff
@@ -173,6 +172,83 @@ PartitionHandleStartDevice(
     return STATUS_SUCCESS;
 }
 
+/**
+ * @brief
+ * Notifies MountMgr to delete all mount points
+ * associated with the given volume.
+ *
+ * @note    This should belong to volmgr.sys and act on a PVOLUME_EXTENSION.
+ **/
+static
+CODE_SEG("PAGE")
+NTSTATUS
+VolumeDeleteMountPoints(
+    _In_ PPARTITION_EXTENSION PartExt)
+{
+    NTSTATUS Status;
+    ULONG InputBufferLength;
+    UNICODE_STRING MountMgr;
+    PUNICODE_STRING DeviceName;
+    PDEVICE_OBJECT DeviceObject;
+    PFILE_OBJECT FileObject = NULL;
+    PMOUNTMGR_MOUNT_POINT InputBuffer = NULL;
+    PMOUNTMGR_MOUNT_POINTS OutputBuffer = NULL;
+
+    /* Setup the volume device name for deleting its mount points */
+    DeviceName = &PartExt->DeviceName;
+
+    /* Allocate the input buffer for MountMgr */
+    InputBufferLength = sizeof(MOUNTMGR_MOUNT_POINT) + DeviceName->Length;
+    InputBuffer = ExAllocatePoolZero(PagedPool, InputBufferLength, TAG_PARTMGR);
+    if (!InputBuffer)
+        return STATUS_INSUFFICIENT_RESOURCES;
+
+    /* Fill it in */
+    InputBuffer->DeviceNameOffset = sizeof(MOUNTMGR_MOUNT_POINT);
+    InputBuffer->DeviceNameLength = DeviceName->Length;
+    RtlCopyMemory(&InputBuffer[1], DeviceName->Buffer, DeviceName->Length);
+
+    /* Allocate big enough output buffer (we don't care about the output).
+     * NOTE: We could do a first call to IOCTL_MOUNTMGR_DELETE_POINTS to
```

Comment:
You could first allocate a tiny buffer and then increase its size like this https://github.com/disean/reactos/blob/ideport/drivers/storage/ide/atapi/ataacpi.c#L202

Reply:
Thanks for reporting this. Also, this function may return STATUS_BUFFER_OVERFLOW twice... I think Driver Verifier can catch such memory bugs

Reply:
I've updated my PR with code adapted from your suggestion; please recheck.